### PR TITLE
feat(draft-pr): add new option to not process PRs which are in draft

### PR DESCRIPTION
**Pull request by C0ZEN**
## Changes

- [x] Add new option "exempt-draft-pr" to not process a PR which is in draft

## Context

Closes https://github.com/actions/stale/issues/36 (even if already closed).
See https://github.com/actions/stale/issues/36#issuecomment-889249688.

### CODE REVIEW

**Comment by derek-bisson-ukg** [comment]
very much looking forward to this!

**Comment by C0ZEN** [comment]
up

File: src/classes/issues-processor.ts [hunk 1]
```diff
@@ -804,34 +836,14 @@ export class IssuesProcessor {
...
-  }
-
   // Delete the branch on closed pull request
   private async _deleteBranch(issue: Issue): Promise<void> {
     const issueLogger: IssueLogger = new IssueLogger(issue);
 
     issueLogger.info(`Delete branch from closed $$type - ${issue.title}`);
 
-    const pullRequest = await this._getPullRequest(issue);
+    const pullRequest: IPullRequest | undefined | void =
```
Comment (luketomlinson) [inline src/classes/issues-processor.ts:845]: I don't think you have to specify the type here. Could just do `const pullRequest = await this._getPullRequest(issue)`

**Review by luketomlinson — APPROVED** [review]
💥 I like it @C0ZEN. Left one minor comment. There's some merge conflicts too.

File: src/classes/issues-processor.ts (hunk 1 above)
Comment (C0ZEN) [reply src/classes/issues-processor.ts:845]: FYI: I do this on purpose because I prefer explicit types; easier to review and code IMO

**Comment by C0ZEN** [comment]
Ready

**Comment by nantiferov** [comment]
@C0ZEN thank you!

@luketomlinson main branch hasn't tagged yet with new release, last is v4.0.0 https://github.com/actions/stale/releases/tag/v4.0.0
Do you plan to update it? I would like to test this feature, but I don't want to use `@main` in GitHub Action

**Pull request merged**